### PR TITLE
fix: modified the frontend

### PR DESCRIPTION
**Pull request by DebarghyaPal**
Customised the Home Page, Login Portal and Signup Page

### CODE REVIEW

File: templates/student/login.html [hunk 1]
```diff
@@ -1,10 +1,105 @@
 {% extends 'base.html' %}
 
-{% block heading %} Login {% endblock heading %}
+{% block heading %} {% endblock heading %}
```
Comment (ArkaprabhaChakraborty) [inline templates/student/login.html:3]: Why was this removed?

**Comment by DebarghyaPal** [comment]
image

**Comment by DebarghyaPal** [comment]
image

**Comment by DebarghyaPal** [comment]
image

**Comment by DebarghyaPal** [comment]
image

File: templates/student/login.html [hunk 2]
```diff
@@ -23,11 +118,11 @@
       type="password"
       placeholder="password"
     />
-  </div>
-  {% if request.GET.next %}
+  </div> {% endcomment %}
+  {% comment %} {% if request.GET.next %}
```
Comment (ArkaprabhaChakraborty) [inline templates/student/login.html:122]: Why were these comment templates added? As far as I know, these are [Liquid](https://shopify.github.io/liquid/). Did you use it? If so how did you manage the requirements for it?

File: templates/student/signup.html [hunk 3]
```diff
@@ -1,11 +1,67 @@
 {% extends 'base.html' %}
-{% block heading %} Signup {% endblock heading %}
+{% block heading %} 
+
+{% endblock heading %}
 
 {% block content %}
 
-<form class="w-2xl sm:w-1/2 max-w-xs flex flex-col mx-auto -mt-8 sm:-mt-12" action="/student/signup/" method="POST">
```
Comment (ArkaprabhaChakraborty) [inline templates/student/signup.html:6]: Screenshot required to assert the correct behaviour

File: templates/student/signup.html [hunk 4]
```diff
@@ -1,11 +1,67 @@
...
+    <label class="text-lg text-gray-500 font-bold" for="password">Password :</label>
+    <input
+      class="w-full sm:w-2/3 px-2 py-1 bg-gray-100 border-gray-300 border-2 mt-4 sm:mt-0 sm:ml-10 rounded text-black"
+      name="password"
+      type="password"
+      placeholder="password"
+    />
+  </div>
+
+  {% comment %} <div class="w-1/2  max-w-xs flex flex-col sm:flex-row mt-4  sm:items-center"></div>
```
Comment (ArkaprabhaChakraborty) [inline templates/student/signup.html:64]: Again ... liquid blocks out here...

**Comment by ArkaprabhaChakraborty** [comment]
Create a student account and an admin account and provide an ss for those too. After the creation of the account issue a book and provide those SS too. I need ss to be certain that everything works fine.

File: templates/student/signup.html [hunk 5]
```diff
@@ -44,18 +100,19 @@
     type="password"
     placeholder="password"
   />
-</div>
-<div class="w-1/2 max-w-xs flex flex-col sm:flex-row mt-4 sm:items-center"></div>
-<label class="text-xl" for="department">Department</label>
-<select class="rounded border border-red-300 py-2 focus:outline-none focus:ring-2 focus:ring-red-200 mt-2 focus:border-red-500 text-base pl-3 pr-10 text-gray-800" name="department">
-  <option   value="">Select a dept</option>
+</div> {% endcomment %}
```
Comment (ArkaprabhaChakraborty) [inline templates/student/signup.html:103]: Liquid statements ... need ss as a proof to be certain that everything works fine

**Review by ArkaprabhaChakraborty — CHANGES_REQUESTED** [review]
Some necessary proofs are required so I can merge them into the main branch :). There can be multiple breaking changes so SS proofs are essential that we don't break essential functionalities.

**Comment by ArkaprabhaChakraborty** [comment]
> image

Does the issue now on trending books work as expected? Does it redirect to the sign-in page if the user isn't logged in? If not the functionality changes have to be made in views.

**Comment by DebarghyaPal** [comment]
image

**Comment by DebarghyaPal** [comment]
While trying to issue a book, it says, "You are not a student."

File: templates/student/login.html (hunk 2 above)
Comment (DebarghyaPal) [reply templates/student/login.html:122]: > Why were these comment templates added? As far as I know, these are [Liquid](https://shopify.github.io/liquid/). Did you use it? If so how did you manage the requirements for it?

I commented out the code snippets which I did not require. You can remove those from the file.

File: templates/student/signup.html [hunk 6]
```diff
@@ -1,11 +1,67 @@
 {% extends 'base.html' %}
-{% block heading %} Signup {% endblock heading %}
```
[the remark this answers is not in the document]
Comment (DebarghyaPal) [reply templates/student/signup.html:2]: > Again ... why was this removed?

It wasn't looking great.

File: templates/student/login.html [hunk 7]
```diff
@@ -1,10 +1,105 @@
 {% extends 'base.html' %}
 
-{% block heading %} Login {% endblock heading %}
+{% block heading %} {% endblock heading %}
 
 
 {% block content %}
-<form class="w-2xl sm:w-1/2 max-w-xs flex flex-col mx-auto " action="/student/login/" method="POST">
```
[the remark this answers is not in the document]
Comment (DebarghyaPal) [reply templates/student/login.html:7]: > Why was this removed? Does the functionality remian the same for the form? Have you verified? If so can you attach a screenshot showing that it works as expected?

Yes, no alteration was done with the functionality.

File: templates/student/login.html (hunk 1 above)
Comment (DebarghyaPal) [reply templates/student/login.html:3]: > Why was this removed?

Wasn't looking great.

**Comment by ArkaprabhaChakraborty** [comment]
I guess the student must be cteated properly and the student signup criteria has to be given by the admin first. Since that feature exists and we have asserted the correct usage I think the PR is good to be merged once we have all SS ready.

**Comment by DebarghyaPal** [comment]
Screenshot (1106)
Screenshot (1107)
Screenshot (1108)
Screenshot (1109)
Screenshot (1110)
Screenshot (1111)
Screenshot (1112)
Screenshot (1113)
Screenshot (1114)
Screenshot (1115)
Screenshot (1116)
Screenshot (1117)
Screenshot (1118)
Screenshot (1119)
Screenshot (1120)
Screenshot (1121)
Screenshot (1122)
Screenshot (1123)
Screenshot (1124)
Screenshot (1125)
Screenshot (1126)
Screenshot (1127)
Screenshot (1128)
Screenshot (1129)

**Pull request merged**